components/sync15/src/state.rs

```diff
@@ -264,34 +107,35 @@ fn new_global_from_previous(
     })
 }
 
-pub struct SetupStateMachine<'client, 'keys> {
+pub struct SetupStateMachine<'client, 'keys, 'appstate> {
```

Comment:
Not sure we need three lifetime params here. I think we actually only need one since none are &mut.